corgidrp/pump_trap_calibration.py

```diff
@@ -0,0 +1,3281 @@
+    check.real_nonnegative_scalar(pc_max, 'pc_max', TypeError)
+    if pc_max <= pc_min:
+        raise ValueError('pc_max must be > pc_min')
+    check.real_nonnegative_scalar(offset_min, 'offset_min', TypeError)
+    check.real_nonnegative_scalar(offset_max, 'offset_max', TypeError)
+    check.real_nonnegative_scalar(cs_fit_thresh, 'cs_fit_thresh', TypeError)
+    if cs_fit_thresh > 1:
+        raise ValueError('cs_fit_thresh should fall in (0,1).')
+    check.real_nonnegative_scalar(E_min, 'E_min', TypeError)
+    check.real_nonnegative_scalar(E_max, 'E_max', TypeError)
+    if E_max <= E_min:
+        raise ValueError('E_max must be > E_min')
+    check.real_nonnegative_scalar(cs_min, 'cs_min', TypeError)
+    check.real_nonnegative_scalar(cs_max, 'cs_max', TypeError)
+    if cs_max <= cs_min:
+        raise ValueError('cs_max must be > cs_min')
+    check.positive_scalar_integer(bins_E, 'bins_E', TypeError)
+    check.positive_scalar_integer(bins_cs, 'bins_cs', TypeError)
+    check.real_positive_scalar(input_T, 'input_T', TypeError)
+
+    # to count how many apparent dipoles that
+    #didn't meet the fit threshold
+    bad_fit_counter = 0
+    # to count how many fit attempts occured
+    num_attempted_fits = 0
+    # counter for number of traps before filtering them for sub-el location
+    # (over all schemes and temperatures)
+    pre_sub_el_count = 0
+    unused_fit_data = None
+    # count # times Pc is bigger than one (relevant when trap_fit_const() used)
+    pc_bigger_1 = 0
+    pc_biggest = 0
+    # count # times emit time constant > 1e-2 or < 1e-6. Relevant only if
+    # tau_min and tau_max are outside of 1e-6 to 1e-2.
+    tau_outside = 0
+    
+    temps = {}
+
+    #First we'll sort the input dataset by temperature
+    dataset_list, dataset_temperatures = working_dataset.split_dataset(exthdr_keywords = ['EXCAMT'])
+    
+    # for dir in os.listdir(base_dir):
+    for i,dataset in enumerate(dataset_list):
+  
+        curr_temp = dataset_temperatures[i]
+        schemes = {}
+        # initializing eperdn here; used if scheme is 1
+        eperdn = 1
+
+        scheme_header_keywords = ['TPSCHEM1','TPSCHEM2', 'TPSCHEM3','TPSCHEM4']
+        scheme_datasets, scheme_list = dataset.split_dataset(exthdr_keywords = scheme_header_keywords)
+        
+        sch_list = []
+        for i in range(len(scheme_datasets)):
+            #Grab the first file's extension header from each dataset
+            header0 = scheme_datasets[i][0].ext_hdr
+
+            #Find the TPSCHEM keyword that is non-zero
+            this_num_pumps = [header0[x] for x in scheme_header_keywords]
+            this_num_pumps = np.array(scheme_list[i] )
+            this_scheme = int(np.where(this_num_pumps != 0)[0]) + 1
+            sch_list.append(this_scheme)
+
+            #Grab the number of pumps from the first dataset. 
+            if i == 0: 
+                num_pumps = this_num_pumps[this_scheme-1]
```

Comment:
Sorry I didn't catch this before.  As I work on the e2e based on this branch, I go a bit deeper into it.  In theory, the number of pumps could be different for each scheme, we shouldn't assume `num_pumps` is the same number for all schemes.

Reply:
makes sense! Fixed now!